tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -274,48 +274,45 @@ def access_string_vhdl(dut):
 # TODO: add tests for Verilog "string_input_port" and "STRING_LOCALPARAM" (see issue #802)
 
 @cocotb.test(skip=cocotb.LANGUAGE in ["vhdl"] or cocotb.SIM_NAME.lower().startswith("icarus"),
-             expect_fail=cocotb.SIM_NAME.lower().startswith("modelsim"),
-             expect_error=cocotb.SIM_NAME.startswith(("xmsim", "ncsim")))
+             expect_error=cocotb.result.TestFailure if cocotb.SIM_NAME.lower().startswith(("xmsim", "ncsim", "modelsim", "chronologic simulation vcs")) else False)
 def access_const_string_verilog(dut):
     """Access to a const Verilog string."""
     tlog = logging.getLogger("cocotb.test")
-
-    yield Timer(10)
     string_const = dut.STRING_CONST;
+
+    yield Timer(10, 'ns')
     tlog.info("%r is %s" % (string_const, str(string_const)))
     if not isinstance(string_const, StringObject):
         raise TestFailure("STRING_CONST was not StringObject")
-    if string_const != "TESTING_CONST":
-        raise TestFailure("STRING_CONST was not == \'TESTING_CONST\'")
+    if string_const != b"TESTING_CONST":
+        raise TestFailure("Initial value of STRING_CONST was not == b\'TESTING_CONST\' but {} instead".format(string_const))
```

Comment:
Or perhaps:
```suggestion
    expected = b"TESTING_CONST"
    if string_const != expected:
        raise TestFailure("Initial value of STRING_CONST was not == {} but {} instead".format(expected, string_const))
```

Reply:
I started this change but then discovered that there are quite a number of such duplications (e.g. also for object types) in the test so that I'll skip this for now.
If ``expected`` would be used a third time I would not hesitate to make this kind of change, but the additionally required mental backtracking for the reader is diminishing the positive keeping-in-sync effect here IMO.

Reply:
Fair enough. This is where the pytest assertion rewriters really provide value.